src/pfe/portal/routes/projects/remoteBind.route.js

```diff
@@ -212,24 +212,23 @@ router.post('/api/v1/projects/:id/upload/end', async (req, res) => {
     const project = user.projectList.retrieveProject(projectID);
     if (project) {
       const pathToTempProj = path.join(global.codewind.CODEWIND_WORKSPACE, global.codewind.CODEWIND_TEMP_WORKSPACE, project.name);
+      const tempProjectExists = await fs.pathExists(pathToTempProj);
       // eslint-disable-next-line no-sync
-      if (!fs.existsSync(pathToTempProj)) {
-        log.info("Temporary project directory doesn't exist, not syncing any files");
-        res.status(404).send("No files have been synced");
+      if (modifiedList.length === 0 && !tempProjectExists) {
+        log.info('Temporary project directory doesn\'t exist and modified list is empty, not syncing any files');
+        res.status(404).send('No files have been synced');
       } else {
-
-        const currentFileList = await listFiles(pathToTempProj, '');
-
-        const filesToDeleteSet = new Set(currentFileList);
-        keepFileList.forEach((f) => filesToDeleteSet.delete(f));
-        const filesToDelete = Array.from(filesToDeleteSet);
+        const pathToProj = path.join(global.codewind.CODEWIND_WORKSPACE, project.name);
```

Comment:
Should you use project.projectPath() here? I think there may be cases the project directory doesn't match the name. They are certainly different fields in the project object.